presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java

```diff
@@ -1328,6 +1329,25 @@ public void testRenameColumn()
         assertStatement("ALTER TABLE foo.t RENAME COLUMN a TO b", new RenameColumn(QualifiedName.of("foo", "t"), identifier("a"), identifier("b")));
     }
 
+    @Test
+    public void testAnalyze()
+    {
+        QualifiedName table = QualifiedName.of("foo");
+        assertStatement("ANALYZE foo", new Analyze(table, ImmutableList.of()));
+
+        assertStatement("ANALYZE foo WITH ( \"string\" = 'bar', \"long\" = 42, computed = 'ban' || 'ana', a = ARRAY[ 'v1', 'v2' ] )",
+                new Analyze(table, ImmutableList.of(
+                        new Property(new Identifier("string"), new StringLiteral("bar")),
+                        new Property(new Identifier("long"), new LongLiteral("42")),
+                        new Property(
+                                new Identifier("computed"),
+                                new FunctionCall(QualifiedName.of("concat"), ImmutableList.of(new StringLiteral("ban"), new StringLiteral("ana")))),
```

Comment:
The test for `||` is brittle. The fact that the parser currently turns it into a FunctionCall for the "concat" function is a legacy implementation detail that's likely to change. It'd be better to test with something more stable, like arithmetic expressions.